FlickrViewer/SearchViewController.swift

```diff
@@ -161,15 +220,54 @@ extension SearchViewController: UICollectionViewDataSource, UICollectionViewDele
         if requestIsFinished {
             if scrollView.contentOffset.y > scrollView.contentSize.height - scrollView.frame.height {
                 self.activityIndicator.startAnimating()
+                if currentSearch != nil, requestIsFinished {
+                    print("Loading next page. Current page now is \(currentPage)")
+                    flickrPhotosSearch(searchText: currentSearch!, pageNumber: currentPage + 1) {
+                    self.justifiedSizes = self.calculateJustifiedSizes(photos: self.photos)
+                    self.collectionView.reloadData()
+                    print("One more page loaded. CURRENT PAGE IS \(self.currentPage)")
+                    print("\(self.justifiedSizes.count) PHOTOS SEARCHED")
+                    self.activityIndicator.stopAnimating()
+                    }
+                } else {
                 getExploreFlickrPhotos(pageNumber: currentPage + 1) {
                     self.justifiedSizes = self.calculateJustifiedSizes(photos: self.photos)
                     self.currentPage += 1
                     print("One more page loaded. CURRENT PAGE IS \(self.currentPage)")
                     print("PHOTOS ARRAY COUNT IS \(self.justifiedSizes.count)")
                     self.collectionView.reloadData()
                     self.activityIndicator.stopAnimating()
```

Comment:
you can extract getExploreFlickrPhotos and flickrPhotosSearch closures in one method